src/core_functions/scalar/list/list_reduce.cpp

```diff
@@ -169,10 +171,22 @@ void LambdaFunctions::ListReduceFunction(duckdb::DataChunk &args, duckdb::Expres
 
 		end = ExecuteReduce(loops, execute_info, info, result_chunk);
 		spare_result_chunk.Reset();
-
 		loops++;
 	}
 
+	// Propagate NULL rows in STRUCTs.
+	if (info.result.GetType().id() == LogicalTypeId::STRUCT) {
```

Comment:
This needs to be recursive in case of structs within structs - perhaps we can use `FlatVector::SetNull` instead that handles this correctly already?

Could we also add a test for nested structs?